reinforcement_learning/common/sagemaker_rl/ray_launcher.py

```diff
@@ -206,8 +213,16 @@ def copy_checkpoints_to_model_output(self):
             print("Saved the checkpoint file %s as %s" % (source_path, destination_path))
 
     def save_experiment_config(self, config):
-        with open(os.path.join(MODEL_OUTPUT_DIR, "params.json"), "w") as f:
-            json.dump(config, f, indent=2)
+        config_found = False
+        for root, directories, filenames in os.walk(INTERMEDIATE_DIR):
+            if config_found:
+                break
+            else:
+                for filename in filenames:
+                    if filename == "params.json":
+                        source = os.path.join(root, filename)
+                        config_found = True
```

Comment:
Why not break after `config_found` but in a different branch?

Reply:
The original code simply dump the provided config into a JSON file. Here we simply ignore the provided config?

Reply:
> Why not break after `config_found` but in a different branch?

This is to break the outer `for` loop `for root, directories, filenames in os.walk(INTERMEDIATE_DIR):`

>The original code simply dump the provided config into a JSON file. Here we simply ignore the provided config?

Yes. The original code breaks if any part of the config is not JSON serializable (e.g., customer policy). We use the config that RLlib generates instead.